### PR TITLE
Add latency and pre fetch method extraction

### PR DESCRIPTION
**Pull request by avinoamkugler**
Introduces 2 small changes:
1. Add server latency collection properties for metrics collections
2. Extract several of the pre-fetch flows introduced in 0.0.11 to allow more flexible usage and init of prefetching in `TongaClient` usages

Bump to ver 0.0.12 to distribute new changes

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -209,6 +210,7 @@ def test_retry_upon_get_exception(self):
             client = TongaClient(server_url, options=TongaClientOptions(retry_delay=0.01, retries=4))
             with self.assertRaises(requests.exceptions.ConnectionError):
                 client.get("flag_name")
+                self.assertAlmostEquals(0.44, client.accumulated_latency, delta=0.1)
```
Comment (eitany) [inline tests/test_client.py]: How does 4 * 0.01 gets to 0.44?

File: tonga/client.py [hunk 2]
```diff
@@ -127,6 +155,7 @@ def _get_from_server_with_retries(self, request_string, headers):
         :rtype: dict or None
         """
         for attempt in range(self.options.retries + 1):
+            start = time()
```
Comment (eladsulami) [inline tonga/client.py]:
```suggestion
            start_time = time()
```

File: tonga/client.py [hunk 3]
```diff
@@ -139,6 +168,8 @@ def _get_from_server_with_retries(self, request_string, headers):
                 if attempt == self.options.retries:
                     raise
                 sleep(self.options.retry_delay)
+            finally:
+                self._time_spent_fetching_from_server += time() - start
```
Comment (eladsulami) [inline tonga/client.py]:
```suggestion
                self._time_spent_fetching_from_server += time() - start_time
```

**Review by eladsulami — APPROVED** [review]

**Review by eitany — APPROVED** [review]

File: tests/test_client.py (hunk 1 above)
Comment (avinoamkugler) [reply tests/test_client.py]: You are right, my bad. it was clocked by the `assertRaises` and not called

**Pull request merged**